cuisine/application/get_cuisines.go

```diff
@@ -0,0 +1,35 @@
+package application
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Abuzar-JS/go-spoonacular-api/cuisine/domain"
+	cuisine "github.com/Abuzar-JS/go-spoonacular-api/cuisine/domain/cuisine"
+)
+
+type GetCuisines func(ctx context.Context) ([]domain.Cuisine, error)
+
+func NewGetCuisines(
+	CuisineRepo cuisine.Repository,
+) GetCuisines {
+	return func(ctx context.Context) ([]domain.Cuisine, error) {
+
+		result, err := CuisineRepo.GetAll(ctx)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to get cuisines")
+		}
+
+		var allCuisines []domain.Cuisine
+
+		for _, value := range result {
```

Comment:
why iterate result into the domain here? GetAll function return the domain value. by using ToDomain function .